marlowe-playground-client/src/Simulation.purs

```diff
@@ -583,44 +576,46 @@ transactionComposer state =
     f (Tuple k v) acc = (Tuple k v) : acc
 
   actionsForParties :: Map Party (Map ActionInputId ActionInput) -> Array (HTML p Action)
-  actionsForParties m = map (\(Tuple k v) -> participant isEnabled k (vs v)) (parties (kvs m))
+  actionsForParties m = map (\(Tuple k v) -> participant state isEnabled k (vs v)) (parties (kvs m))
 
 participant ::
   forall p.
+  State ->
   Boolean ->
   Party ->
   Array ActionInput ->
   HTML p Action
-participant isEnabled party actionInputs
+participant state isEnabled party actionInputs
   | party == otherActionsParty =
     li [ classes [ ClassName "participant-a", noMargins ] ]
       ( [ h6_ [ em_ [ text "Other Actions" ] ] ]
-          <> (map (inputItem isEnabled (partyName otherActionsParty)) actionInputs)
+          <> (map (inputItem state isEnabled (partyName otherActionsParty)) actionInputs)
       )
     where
     partyName (PK name) = name
 
     partyName (Role name) = name
 
-participant isEnabled (PK person) actionInputs =
+participant state isEnabled (PK person) actionInputs =
```

Comment:
This code blocks are sooo similar it makes me wonder if you've got the pattern match on `Party` in the right place.

Reply:
refactored